speedy-links/src/speedy-links.ts

```diff
@@ -0,0 +1,99 @@
+export interface ActivateLinksOptions {
+  mapLinkUrlToModuleUrl: (url: string) => string | Promise<string>;
+  linkSelector?: string;
+  moduleExecutor?: (module: any) => Promise<void>;
+}
+
+let hasBeenSetup = false;
+let opts: ActivateLinksOptions;
+let linkSelector: string;
+const modulesCache = {};
+
+export function setupSpeedyLinks(options: ActivateLinksOptions): void {
+  if (hasBeenSetup) {
+    return;
+  }
+  hasBeenSetup = true;
+  opts = options;
```

Comment:
I dont get this reference holding. Why not just destructure the opts parameters and even setup default value for link selector